initialize-form.md from Reza. (#81)
plantanapp/documentation: docs/actions/initialize-form.md

The page will not be refreshed when a Form is initialized.
:::

## `Actions Covered by this Document`

| Action | Purpose |
| -- | -- |
| Initialize Action Form And Continue Execution | Initializes the specified Action Form and then continues to subsequent actions on the button and submission. |
| Initialize Action Form And Stop Execution | Opens the specified Action Form and then stops further execution of both subsequent actions and HTTP post submission. |

## `Typical Use Cases`

- Show a Form that is not visible by default
- Re-initialze a form that was already submitted, but has not been closed so multiple entries can be submitted

## `Don't use it to`

- Open a Form in a pop-up. Use the Open Form Popup action instead.
  
## `Related Actions`

| Action Name                                       | Description                                                                                       |
| ------------------------------------------------- | ------------------------------------------------------------------------------------------------- |
| [Open Form Popup](/docs/actions/open-form-popup)       | Opens a specified Form in a pop-up.|

## `Input Parameter Reference`

> 

Author: ```suggestion

The parameters unique to this action are listed below. [Review the common parameters for all actions here](/docs/actions/common-parameters).

```

| Parameter                  | Description                                                                                                                                                                | Supports Tokens | Default            | Required |
| -------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------- | ------------------ | -------- |
| Description            | A short description for the action. Only Citizen Developers and Admins will be able to see this field. | No              | `empty string` | No      |
| Error Message                  | A generic error message that will be displayed to end users in case an error occurs in the action.  Citizen Developer and above will also see the detailed error message provided by the system.  Leave empty to use the default message.                                                                                                 | Yes             | `empty string`     | No      |

> | Condition                   | This boolean expression is used to determine if this action will execute. Use it to enable or disable this action programatically. For example, you’d enable a ShowError action only if you’ve found an error let’s say when you parsed a response from a web service. A common example is [SomeField] == “Some Value”.                                                                 | Yes             | `empty string`     | No       |

Reviewer: Sounds good.  Should we at least link to that section from here?

> | Condition                   | This boolean expression is used to determine if this action will execute. Use it to enable or disable this action programatically. For example, you’d enable a ShowError action only if you’ve found an error let’s say when you parsed a response from a web service. A common example is [SomeField] == “Some Value”.                                                                 | Yes             | `empty string`     | No       |

Author: I added a suggested line for a link to the common parameters above.

| Select Module | Select the Form module that you wish to initializeby selecting it from a prepopulated list of all the Forms on the page. Alternativly you can set it programatically by switching to an expression to use My Tokens or values based on other fields.                                                                                                                      | Yes              | `None selected`    | Yes      |

> | QueryString Parameters                 | Optionally enter parameters as Name/Value pairs that can be accessed through querystring. The values set here will be sent to the Form module and will be available in the Init and PreInit actions as values of the 'QueryString:' token namespace.                                                                                                                                    | Yes              | `None`               | No       |

Author: ```suggestion
| QueryString Parameters                 | Optionally, enter parameters as `Name` and `Value` pairs that will be passed to the form being initialized. The values set here will be sent to the Form module and will be available in the Init and PreInit actions using the `[QueryString:Parameter]` token syntax.                                                                                                                                   | Yes              | `None`               | No       |
```

|

> ## `Output Parameters Reference`

Author: I have been leaving this section out if there aren't any output params. Do you think we should always have it in place?

Reviewer: We can leave it out.  Thought it would be good to keep a consistent structure and implicitly state that it doesn't apply so people know there is no output token.  In this case, the consistent structure would be to leave it out since that's what we're already doing :-)

- Open a Form in a pop-up. Use the Open Form Popup action instead.
  
## `Related Actions`

| Action Name                                       | Description                                                                                       |
| ------------------------------------------------- | ------------------------------------------------------------------------------------------------- |
| [Open Form Popup](/docs/actions/open-form-popup)       | Opens a specified Form in a pop-up.|

## `Input Parameter Reference`

The parameters unique to this action are listed below. [Review the common parameters for all actions here](/docs/actions/common-parameters).

| Parameter                  | Description                                                                                                                                                                | Supports Tokens | Default            | Required |
| -------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------- | ------------------ | -------- |

> | Description            | A short description for the action. Only Citizen Developers and Admins will be able to see this field. | No              | `empty string` | No      |
> | Error Message                  | A generic error message that will be displayed to end users in case an error occurs in the action.  Citizen Developer and above will also see the detailed error message provided by the system.  Leave empty to use the default message.                                                                                                 | Yes             | `empty string`     | No      |
> | Condition                   | This boolean expression is used to determine if this action will execute. Use it to enable or disable this action programatically. For example, you’d enable a ShowError action only if you’ve found an error let’s say when you parsed a response from a web service. A common example is [SomeField] == “Some Value”.                                                                 | Yes             | `empty string`     | No       |

Author: ```suggestion
```
